server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java

```diff
@@ -390,6 +397,40 @@ public boolean deleteProjectMember(@NonNull Long projectId, @NonNull Long userId
         return deleted;
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    @SkipAuthorize("Internal usage")
+    public Boolean batchDelete(Set<Long> projectIds) {
+        projectPermissionValidator.checkProjectRole(projectIds, Collections.singletonList(ResourceRoleName.OWNER));
+        repository.findByIdIn(projectIds).stream().forEach(project -> {
+            if (project.getBuiltin() || !project.getArchived()) {
+                throw new UnsupportedException(ErrorCodes.IllegalOperation,
+                        new Object[] {"builtin or not-archived project, projectName=" + project.getName()},
+                        "Operation on builtin or not-archived project is not allowed");
+            }
+        });
+        repository.deleteAllById(projectIds);
```

Comment:
can we collect all resource needed be deleted in one place, or we can bind some tag to it. cause if someone  introduce some new feature,  the coder may not aware he need add delete code here. 
anyway this is not block issue.

Reply:
Yes, I will add a TODO and optimize it later for not blocking this PR.

Reply:
Currently, the database change templates is also bound with project. Those may to be deleted together when deleting the project